src/Microsoft.AspNetCore.Http.Abstractions/QueryString.cs

```diff
@@ -209,7 +215,10 @@ public QueryString Add(string name, string value)
             builder.Append("&");
             builder.Append(UrlEncoder.Default.Encode(name));
             builder.Append("=");
-            builder.Append(UrlEncoder.Default.Encode(value));
```

Comment:
You can also use `AppendKeyValuePair` with `first = false` here.

Reply:
Agreed. Committed.